pkg/rabbit/queue.go

```diff
@@ -79,10 +79,31 @@ func NewPolicy(args *QueueArgs) *rabbitv1beta1.Policy {
 		},
 		Spec: v1beta1.PolicySpec{
 			Name:                     args.Name,
+			Priority:                 1,
 			Pattern:                  fmt.Sprintf("^%s$", regexp.QuoteMeta(args.Name)),
 			ApplyTo:                  "queues",
 			Definition:               &runtime.RawExtension{Raw: []byte(fmt.Sprintf(`{"dead-letter-exchange": %q}`, *args.DLXName))},
 			RabbitmqClusterReference: *args.RabbitmqClusterReference,
 		},
 	}
 }
+
+// NewBrokerDLXPolicy configures the broker dead letter exchange for trigger queues that does not have dlx defined
+func NewBrokerDLXPolicy(args *QueueArgs) *rabbitv1beta1.Policy {
+	return &rabbitv1beta1.Policy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            args.Name,
+			Namespace:       args.Namespace,
+			OwnerReferences: []metav1.OwnerReference{args.Owner},
+			Labels:          args.Labels,
+		},
+		Spec: v1beta1.PolicySpec{
+			Name:                     args.Name,
+			Priority:                 0, // lower priority then policies created for trigger queues to allow overwrite
+			Pattern:                  fmt.Sprintf("^%s.", regexp.QuoteMeta("t.q.")),
```

Comment:
This pattern doesn't distinguish based on Broker name. If we have 2 brokers with delivery spec defined, it looks like they'll create two policies with identical patterns.

Reply:
Oh ok ok, and avoid edge cases where in broker name is a subset of other

Reply:
Think the problem would exist even if one name wasn't the subset of the other. The pattern used for the policy is a static `^t.q.` which all trigger queues will fall under

Reply:
Yeah I ment with the broker name prefix fix